src/temporal/t.rast.climatologies/t.rast.climatologies.py

```diff
@@ -0,0 +1,244 @@
+            )
+            maps_name = [sam.get_id() for sam in selemaps]
+            # check if there are maps for that day
+            if len(maps_name) > 0:
+                outname = f"{basename}_{doystr}"
+                runmod = copy.deepcopy(mod)
+                runmod.inputs.input = ",".join(maps_name)
+                runmod.outputs.output = outname
+                process_queue.put(runmod)
+                map_layer = tgis.space_time_datasets.RasterDataset(
+                    f"{outname}@{mapset}"
+                )
+                extent = tgis.RelativeTemporalExtent(
+                    start_time=doy - 1,
+                    end_time=doy,
+                    unit=outunit,
+                )
+                map_layer.set_temporal_extent(extent=extent)
+                outmaps.append(map_layer)
+
+            if doy % 10 == 0:
+                gscript.percent(doy, 366, 1)
+    else:
+        outunit = "months"
+        for month in range(1, 13):
+            monthstr = "{:02d}".format(month)
+            thiswhere = f"strftime('%m', start_time) == '{monthstr}'"
+            selemaps = insp.get_registered_maps_as_objects(
+                thiswhere, "start_time", None
+            )
+            maps_name = [sam.get_id() for sam in selemaps]
+            if len(maps_name) > 0:
+                outname = f"{basename}_{monthstr}"
+                runmod = copy.deepcopy(mod)
+                runmod.inputs.input = ",".join(maps_name)
+                runmod.outputs.output = outname
+                process_queue.put(runmod)
+                map_layer = tgis.space_time_datasets.RasterDataset(
+                    f"{outname}@{mapset}"
+                )
+                extent = tgis.RelativeTemporalExtent(
+                    start_time=month - 1,
+                    end_time=month,
+                    unit=outunit,
+                )
+                map_layer.set_temporal_extent(extent=extent)
+                outmaps.append(map_layer)
+            gscript.percent(month, 12, 1)
+
+    if not space_time:
+        # create new space time raster dataset
+        output_strds = tgis.open_new_stds(
+            output,
+            "strds",
+            "relative",
+            f"{gran} {method} climatologies",
+            f"Climatologies created from {strds}, {gran} {method} maps",
+            semantic_type,
+            dbif,
+            gscript.overwrite(),
+        )
+        register_null = False
+        # register maps into space time raster dataset
+        tgis.register_map_object_list(
+            "rast",
+            outmaps,
+            output_strds,
+            register_null,
+            outunit,
+            dbif,
+        )
+
+        # Update the raster metadata table entries
+        output_strds.metadata.update(dbif)
+
+    dbif.close()
+    return True
+
+
+if __name__ == "__main__":
+    import grass.pygrass.modules as pymod
+    import grass.script as gscript
+    import grass.temporal as tgis
```

Comment:
This is not the lazy importing you are looking for. This code will still run before the parser() call which is at the beginning of the main function. You want to put this after that.

Reply:
Now it should be fixed